feat: add new migration guides [netlify-build] (#4783)
segmentio/segment-docs: src/connections/sources/catalog/libraries/mobile/kotlin-android/migration.md

        configuration.setTrackApplicationLifecycleEvents(true);
        return Unit.INSTANCE;
        }
    );

    // Wrap the object with JavaAnalytics for Java Compatibility.
    // You can also choose not to wrap the object, but some of the Analytics methods may not be accessible.
    JavaAnalytics analytics = new JavaAnalytics(androidAnalytics);   
  ```
  {% endcodeexampletab %}
  {% endcodeexample %}

### 1.b) Update your import statements

  {% codeexample %}
  {% codeexampletab Before%}
  ```java
    import com.segment.analytics.Analytics;
    import com.segment.analytics.Middleware;   
  ```
  {% endcodeexampletab %}
  {% codeexampletab After%}
  ```java
    import com.segment.analytics.kotlin.core.Analytics;
    import com.segment.analytics.kotlin.android.AndroidAnalyticsKt; // For calling from Android
    import com.segment.analytics.kotlin.core.compat.JavaAnalytics; // For calling from Java
    import com.segment.analytics.kotlin.core.platform.Plugin; // Replaces Middleware
  ```
  {% endcodeexampletab %}
  {% endcodeexample %}

## 2. Upgrade your Destinations

> If your app uses Segment to route data to Destinations via Segment-cloud (i.e. Cloud-mode destinations), you can skip this step. Analytics-Kotlin treats Device-mode Destinations as plugins, and simplifies the process in integrating them into your app. Analytics-Kotlin supports these Device-Mode Destinations with more to come.

Reviewer: ```suggestion
If your app uses Segment to route data to Destinations through Segment-cloud (for example, Cloud-mode destinations), you can skip this step. Analytics-Kotlin treats Device-mode Destinations as plugins, and simplifies the process in integrating them into your app. Analytics-Kotlin supports these Device-Mode Destinations with more to come.
```

### 2.a) Import the Destination Plugin

```java
    implementation '<owner>:<project>:<version>'
```

### 2.b) Add Plugin to your Analytics instance

  {% codeexample %}
  {% codeexampletab Java%}
  ```java
    analytics.add(new SomeDestinationPlugin());
  ```
  {% endcodeexampletab %}
  {% codeexampletab Kotlin%}
  ```java
    analytics.add(SomeDestinationPlugin())
  ```
  {% endcodeexampletab %}
  {% endcodeexample %}

Your events will now begin to flow to the added destination in Device-Mode.

## 3. Upgrade Middleware to Plugins

> Middlewares are a powerful mechanism that can augment events collected by the Analytics iOS (Classic) SDK. A middleware is a simple function that is invoked by the Segment SDK and can be used to monitor, modify, augment or reject events. Analytics Swift replaces the concept of middlewares with Enrichment Plugins to give you even more control over your event data. Refer to the Plugin Architecture Overview for more information.

Reviewer: @alanjcharles Should this paragraph reference Kotlin?

Author: @markzegarelli oh, yeah definitely

`autoAddSegmentDestination` | The analytics client automatically adds the Segment Destination. Set this to `false`, if you want to customize the initialization of the Segment Destination, such as, add destination middleware). |

<br> Segment removed these options:

Option | Details
------ | --------
`defaultOptions` | Removed in favor of a plugin that adds the default data to the event payloads. Segment doesn't provide a plugin example since it's dependent on your needs.|
`recordScreenViews` | Removed in favor of the `AndroidRecordScreenPlugin` that provides the same functionality. |
`trackAttributionData` | This feature no longer exists. |       
                                        
### 4.b) Properties

Properties have been replace by JsonElement. Since Properties are essentially a map we provide a conversion function
to convert Maps of type Map<String, Object> to JsonElement:

Kotlin
val map : MutableMap<String, Object> = HashMap<String, Object>()

val json = map.toJsonElement()

Java
Map map = new HashMap<String, Object>();
JsonElement json = JSONKt.toJsonElement(map);

### 4.c) Options
Options are no longer supported and should be converted into plugins.
## Conclusion

> Hopefully this simplifies your migration to Analytics-Kotlin. Once you’re up and running, you can take advantage of Analytics-Kotlin’s additional features, such as Destination Filters, Functions & Typewriter support.

Reviewer: ```suggestion
Once you’re up and running, you can take advantage of Analytics-Kotlin’s additional features, like [Destination Filters](/docs/connections/sources/catalog/libraries/mobile/kotlin-android/kotlin-android-destination-filters/), [Functions](https://segment.com/docs/connections/functions/), and [Typewriter](/docs/connections/sources/catalog/libraries/mobile/kotlin-android/kotlin-android-typewriter/) support.
```
